docs/theming-flatpak.rst

```diff
@@ -0,0 +1,56 @@
+Theming Flatpak
+===============
+
+Flatpak apps default to Adwaita as a theme, as they cannot use the system theme. This happens because all flatpaks are sandboxed and do not have the ability to read data files or libraries in ``/usr`` (where system themes are typically located). This is done so that the format is portable across a wide range of distributions in a reproducible environment, but has the downside of not being able to pick up themes. The solution to this was to package GTK themes as Flatpaks, as relying upon the host to have the correct version for every flatpak defeats the portability benefits it provides. These themes are provided as extensions to the Gnome runtime.
+
+Installing themes
+-----------------
+
+The current themes are packaged in the flathub repository which you can add (if it's not already added) by running::
+
+  $ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
+
+To see a list of currently packaged themes you can run::
+
+  $ flatpak remote-ls flathub | grep org.gtk.Gtk3theme
+
+The list as of posting is:
+
+- org.gtk.Gtk3theme.Adapta
+- org.gtk.Gtk3theme.Adapta-Eta
+- org.gtk.Gtk3theme.Adapta-Nokto
+- org.gtk.Gtk3theme.Adapta-Nokto-Eta
+- org.gtk.Gtk3theme.Ambiance
+- org.gtk.Gtk3theme.Arc
+- org.gtk.Gtk3theme.Arc-Dark
+- org.gtk.Gtk3theme.Arc-Dark-solid
+- org.gtk.Gtk3theme.Arc-Darker
+- org.gtk.Gtk3theme.Arc-Darker-solid
+- org.gtk.Gtk3theme.Arc-solid
+- org.gtk.Gtk3theme.Breeze
+- org.gtk.Gtk3theme.Breeze-Dark
+- org.gtk.Gtk3theme.Greybird
+- org.gtk.Gtk3theme.Numix
+- org.gtk.Gtk3theme.Pop
+- org.gtk.Gtk3theme.Pop-light
+- org.gtk.Gtk3theme.elementary
+
+You can install them with ``flatpak install flathub org.gtk.Gtk3theme.Foo``.
+
+Applying themes
+---------------
+
+There is no way to specify the theme Flatpak applications use. The applications will try to match the system theme currently being used, if it corresponds to any of the Flatpak themes installed, and will fall back to Adwaita if a corresponding theme isn't detected. This happens because the ``flatpak`` tool doesn’t depend upon Gtk so it will not automatically detect your theme.
+
+The theming system requires Flatpak 0.8.4+ and applications using up to date ``org.gnome.Platform`` 3.24+.
+
+Other notes on theming
+----------------------
+
+The same solution exists for icon themes and more will be packaged soon. They can be searched with ``flatpak remote-ls flathub grep org.freedesktop.Platform.Icontheme``.
+
+This process will likely be automated by Gnome-Software at some point.
+
+Themes are picked up via GtkSettings, which means you require a settings-daemon running to set this. On GNOME (and GNOME-derived desktop environments) this will already be running but for other desktops you can run ``gsd-xsettings`` directly in your libexec directory provided by ``gnome-settings-daemon``.
```

Comment:
This could use some improvement.

Gtk3 on X11 picks up themes via XSettings. Gtk3 on Wayland picks up themes via DConf. For the former to work you need an xsettings daemon that is correctly configured. For the latter to work you can either use KDE (with `kde-gtk-config` > 5.11.95) or GNOME which work out of the box or manually configure the dconf keys under `/org/gnome/desktop/interface/`